src/sign.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to sign models."""
+
+import argparse
+import logging
+import pathlib
+
+from model_signing import model
+from model_signing.hashing import file
+from model_signing.hashing import memory
+from model_signing.serialization import serialize_by_file
+from model_signing.signature import fake
+from model_signing.signature import key
+from model_signing.signature import pki
+from model_signing.signature import signing
+from model_signing.signature import sigstore
+from model_signing.signing import in_toto
+from model_signing.signing import in_toto_signature
+
+
+log = logging.getLogger(__name__)
+
+
+def _arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser("Script to sign models")
+    parser.add_argument(
+        "--model_path",
+        help="path to the model to sign",
+        required=True,
+        type=pathlib.Path,
+        dest="model_path",
+    )
+    parser.add_argument(
+        "--sig_out",
+        help="the output file, it defaults ./signature.json",
```

Comment:
Why? We were going to use `model.sig` if inside the model directory or `<model_name>.sig` if model is a file. Why change this in the PR?

Reply:
To make it adjustable by the user

Reply:
Thank you for the feedback (although I did say "we were going to...")

We should still change the help string here.

Reply:
Yes, sure sorry.